cli/cmd/stat.go

```diff
@@ -93,6 +135,26 @@ If no resource name is specified, displays stats about all resources of the spec
   linkerd stat services --from deploy/hello1 --from-namespace test --all-namespaces`,
 		Args:      cobra.RangeArgs(1, 2),
 		ValidArgs: util.ValidTargets,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			err := options.validateConflictingFlags()
+			if err != nil {
+				return err
+			}
+
+			canonical, err := resourceType(args[0])
+			if err != nil {
+				return err
+			}
+
+			if canonical == k8s.Namespace {
+				err := options.validateNamespaceFlags()
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
```

Comment:
For the sake of testability, I'd move all of this code into a separate function, and then add tests for that function. Something like:

```
		PreRunE: func(cmd *cobra.Command, args []string) error {
			return options.validate(args)
		},
```

And:

```
func (options *statOptions) validate(resource []string) error {
	err := options.validateConflictingFlags()
	if err != nil {
		return err
	}

	res, err := util.BuildResource(options.namespace, resource...)
	if err != nil {
		return err
	}

	if res.Type == k8s.Namespace {
		return options.validateNamespaceFlags()
	}

	return nil
}
```

(note that this implementation uses `util.BuildResource` as suggested in my other comment)